### PR TITLE
Update versions of several libraries

### PR DESCRIPTION
**Pull request by serras**
This PR updates several libraries:
- Kotlin to 1.8.21,
- Dokka to 1.8.10,
- KotlinX Coroutines to 1.7.0,
- `kotlin-compile-testing` to 1.5.0, aligned also with the compiler.
And more importantly, it uses an Arrow Gradle Config which also updates many of the plug-in, including the SemVer one. In my local tests the configure phase takes now around 30s. on a clean project, and 2s. after the first run.

### CODE REVIEW

**Comment by nomisRev** [comment]
> I'm marking this as draft, because I don't think we should tinker with it before releasing 1.2.0.

Shouldn't we bump Dokka or Kotlin to 1.8.20 (being rolled out as we speak).

**Comment by serras** [comment]
> Shouldn't we bump Dokka or Kotlin to 1.8.20 (being rolled out as we speak)?

We need to wait until KSP and `kotlin-compile-testing` catch up too.

**Comment by nomisRev** [comment]
Not sure about `kotlin-compile-testing` but KSP is already prepared for Kotlin 1.8.20, https://github.com/google/ksp/releases/tag/1.8.20-RC2-1.0.9

**Comment by serras** [comment]
@JavierSegoviaCordoba I've added the plug-in to every project, is this the right way to proceed?

@nomisRev I've bumped Kotlin to 1.8.20, let's see whether `kotlin-compile-testing` keeps working

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 dokka = "1.8.10"
 jUnit = "4.13.2"
-jUnitJupiter = "5.8.2"
-jUnitVintage = "5.8.2"
-kotest = "5.5.4"
-kotestGradle = "5.5.4"
+jUnitJupiter = "5.9.2"
+jUnitVintage = "5.9.2"
+kotest = "5.5.5"
+kotestGradle = "5.5.5"
```
Comment (JavierSegoviaCordoba) [inline gradle/libs.versions.toml:12]: I think this version had issues on native (missing klib)

Comment (serras) [reply gradle/libs.versions.toml]: Indeed. Let's try with the 5.6.x series.

**Review by JavierSegoviaCordoba — APPROVED** [review]

**Comment by nomisRev** [comment]
@serras we seem to have this strange issue on Windows only 🤔

>* What went wrong:
[59](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:61)
Could not determine the dependencies of task ':arrow-annotations:linkDebugTestMingwX64'.
[60](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:62)
> Could not resolve all dependencies for configuration ':arrow-annotations:mingwX64CompileKlibraries'.
[61](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:63)
   > Failed to calculate the value of task ':arrow-annotations:compileJava' property 'javaCompiler'.
[62](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:64)
      > No matching toolchains found for requested specification: {languageVersion=8, vendor=any, implementation=vendor-specific}.
[63](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:65)
         > No locally installed toolchains match (see https://docs.gradle.org/8.1.1/userguide/toolchains.html#sec:auto_detection) and toolchain download repositories have not been configured (see https://docs.gradle.org/8.1.1/userguide/toolchains.html#sub:download_repositories).
[64](https://github.com/arrow-kt/arrow/actions/runs/5154960585/jobs/9284034998?pr=3017#step:4:66)

A quick Google search turned up this: https://youtrack.jetbrains.com/issue/KTIJ-24981/Gradle-8.-project-sync-fails-with-an-error-No-matching-toolchains-found-for-requested-specification-if-there-is-no-necessary-JDK.

Strange this issue only comes up in Windows though 😕

**Comment by serras** [comment]
> A quick Google search turned up this: https://youtrack.jetbrains.com/issue/KTIJ-24981/Gradle-8.-project-sync-fails-with-an-error-No-matching-toolchains-found-for-requested-specification-if-there-is-no-necessary-JDK.
> 
> Strange this issue only comes up in Windows though 😕

I've tried to apply the steps for [toolchain auto-detection](https://docs.gradle.org/8.0.2/userguide/toolchains.html#sub:download_repositories), let's see if that helps.

**Pull request merged**